core/test-api/kafka/src/main/java/org/opennms/core/test/kafka/JUnitKafkaServer.java

```diff
@@ -0,0 +1,171 @@
+    private KafkaServer kafkaServer;
+    private TestingServer zkServer;
+
+    @Override
+    public void before() throws Exception {
+        zkServer = new TestingServer();
+        // Start ZooKeeper, this method will block until the service has started
+        zkServer.start();
+
+        getAvailablePort(kafkaPort, 9192);
+
+        // Delete any existing Kafka log directory
+        FileUtils.deleteDirectory(new File("target/kafka-log"));
+
+        localhost = getLocalhost();
+
+        final Properties properties = new Properties();
+        properties.put("broker.id", "1");
+        properties.put("auto.create.topics.enable", "true");
+        properties.put("num.partitions", "100");
+        properties.put("enable.zookeeper", "true");
+        properties.put("host.name", localhost);
+        properties.put("log.dir", "target/kafka-log");
+        properties.put("port", String.valueOf(kafkaPort.get()));
+        properties.put("zookeeper.connect", zkServer.getConnectString());
+
+        System.err.println("Kafka server properties: " + properties);
+        kafkaConfig = new KafkaConfig(properties);
+
+        final List<KafkaMetricsReporter> kmrList = new ArrayList<>();
+        final Buffer<KafkaMetricsReporter> metricsList = scala.collection.JavaConversions.asScalaBuffer(kmrList);
+        kafkaServer = new KafkaServer(kafkaConfig, new SystemTime(), Option.<String>empty(), metricsList);
+        kafkaServer.startup();
+        await().atMost(1, MINUTES).until(() -> getBrokerMetadatas(), hasSize(greaterThanOrEqualTo(1)));
+
+        System.err.println("Kafka Address: " + getKafkaConnectString());
+        System.err.println("Zookeeper Address: " + getZookeeperConnectString());
+    }
+
+    @After
+    public void after() {
+        if (kafkaServer != null) {
+            kafkaServer.shutdown();
+        }
+
+        if (zkServer != null) {
+            try {
+                zkServer.stop();
+            } catch (IOException e) {
+                LOG.warn("Failed to stop the ZooKeeper server.", e);
+            }
+            zkServer = null;
+        }
+    }
+
+    private List<BrokerMetadata> getBrokerMetadatas() {
+        ZkClient zkClient = new ZkClient(getZookeeperConnectString(), 1000, 1000, ZKStringSerializer$.MODULE$);
+        ZkUtils zkUtils = new ZkUtils(zkClient, new ZkConnection(getZookeeperConnectString()), false);
+        return JavaConversions.asJavaList(AdminUtils.getBrokerMetadatas(zkUtils, Enforced$.MODULE$, Option.empty()));
+    }
+
+    private static int getAvailablePort(final AtomicInteger current, final int max) {
+        while (current.get() < max) {
+            try (final ServerSocket socket = new ServerSocket(current.get())) {
+                return socket.getLocalPort();
+            } catch (final Throwable e) {}
+            current.incrementAndGet();
+        }
+        throw new IllegalStateException("Can't find an available network port");
+    }
+
+    private static String getLocalhost() {
```

Comment:
I was specifically matching the way that Apache Curator sets up localhost because I was getting weird mismatches running the tests on my machine using the InetAddressUtils version.